src/hooks/useSignErrorHandler.ts

```diff
@@ -0,0 +1,62 @@
+import { t } from '@lingui/macro';
+import { useEffect } from 'react';
+
+import useToastMsg from './useToastMsg';
+import type { VCSignError } from './useVerifiableCredential';
+import { VCSignErrorType } from './useVerifiableCredential';
+
+export type CustomErrorProps = {
+  [key: string]: {
+    title: string;
+    description: string;
+  };
+};
+
+/**
+ * A Hook to handle sign error and display the error message in toast.
+ *
+ * @param signError - The sign error.
+ * @param errorMessages - The override error messages.
+ */
+export function useSignErrorHandler(
+  signError: VCSignError | undefined,
+  errorMessages?: CustomErrorProps,
+) {
+  const { showErrorMsg } = useToastMsg();
+
+  useEffect(() => {
+    if (signError) {
+      const defaultTitle = t`Unknown Error`;
+      const defaultDescription = t`Unknown Error`;
+
+      const errorMsg = errorMessages ?? {};
+
+      for (const item in VCSignErrorType) {
+        if (!errorMsg[item]) {
+          if (item === VCSignErrorType.SignError) {
+            errorMsg[item] = {
+              title: t`Failed to sign the message`,
+              description: signError?.message ?? t`Failed to sign the message`,
```

Comment:
`signError` should never be undefined here. TypeScript should detect that automatically, if it doesn't feel free to ignore my comment

```suggestion
              description: signError.message ?? t`Failed to sign the message`,
```

Reply:
you r right, i have done the change.